client/app/widgets/Notifications/index.jsx

```diff
@@ -18,117 +18,121 @@ export type State = {
   };
 
-  getPusherKey = (signedInKey: number) => {
-    this.setState({ signedInKey });
-    const metaPusherKey = Array.from(
-      window.document.getElementsByTagName('meta'),
-    ).filter((item) => item.getAttribute('name') === 'pusher-key')[0];
-    const pusherKey = metaPusherKey.getAttribute('content');
-    const pusher = new window.Pusher(pusherKey, { encrypted: true });
-    const channel = pusher.subscribe(`private-${signedInKey}`);
-    channel.bind('new_notification', (response: any) => {
-      if (response && response.data) {
-        this.fetchNotifications();
-      }
-    });
-  };
-
-  setBody = (notifications: string[]) => {
-    let updatedNotifications = '';
-    notifications.forEach((item: string) => {
-      updatedNotifications += `<div>${item}</div>`;
-    });
-    this.setState({ notifications: updatedNotifications });
-  };
-
-  fetchNotifications = () => {
-    const { alreadyMounted, signedInKey } = this.state;
+  const fetchNotifications = () => {
+    setAlreadyMounted(alreadyMounted);
+    setSignedInKey(signedInKey);
     return axios
       .get('/notifications/signed_in')
       .then((response: any) => {
         if (response && response.data && response.data.signed_in !== -1) {
           if (response.data.signed_in !== signedInKey) {
-            this.getPusherKey(response.data.signed_in);
+            getPusherKey(response.data.signed_in);
           }
           return axios.get('/notifications/fetch_notifications');
         }
         return -1;
       })
       .then((response: any) => {
         if (response && response.data && response.data.fetch_notifications) {
-          this.changeTitle(response.data.fetch_notifications.length);
-          this.setBody(response.data.fetch_notifications);
+          changeTitle(response.data.fetch_notifications.length);
+          setBody(response.data.fetch_notifications);
           if (!alreadyMounted && response.data.fetch_notifications.length > 0) {
-            this.setState({
-              alreadyMounted: true,
-              open: true,
-              modalKey: Utils.randomString(),
-            });
+            setAlreadyMounted(true);
+            setOpen(true);
+            setModalKey(Utils.randomString());
           }
         }
       });
   };
 
-  clearNotifications = () => {
+  useEffect(() => fetchNotifications(), []);
+
+  const getPusherKey = (signedInKey1: number) => {
+    setSignedInKey(signedInKey1);
+    const metaPusherKey = Array.from(
+      window.document.getElementsByTagName('meta'),
+    ).filter((item) => item.getAttribute('name') === 'pusher-key')[0];
+    const pusherKey = metaPusherKey.getAttribute('content');
+    const pusher = new window.Pusher(pusherKey, { encrypted: true });
+    const channel = pusher.subscribe(`private-${signedInKey1}`);
+    channel.bind('new_notification', (response: any) => {
+      if (response && response.data) {
+        fetchNotifications();
+      }
+    });
+  };
```

Comment:
Hey @Manasa2850, I'm brand new, but I think you can fix the `no-use-before-defined` warning you're seeing by moving this declaration (lines 75 - 88) above line 46. 

`no-use-before-defined` is complaining because you're referring to `getPusherKey` on line 46, but defining `getPusherKey` on 75. Does that make sense?

Reply:
That won't help because `getPusherKey()` and `fetchNotifications()` use each other

Reply:
Oh. Thanks. Let me look a little more closely then. I'll post a new comment when I have a slightly better understanding of what's happending

Reply:
Could we have `getPusherKey` return the key from line 80 or the pusher instance on line 81 and pull out a separate function `subscribeToPusher`  with the code on line 82 - 87? Right now it's doing 2 things: getting the key *and* subscribing to the channel with that key. Splitting that into two functions might let us define `getPusherKey` above `fetchNotifications` and add some clarity about the responsibility of each function.

(I don't have the project running locally, or much context for the project, so this might be out of step with  practices in the repo, misinformed, or just bad. I'm sorry if any of those are true).

Reply:
That didn't work too :(
You can check my latest commits. I've tried what you suggested

Reply:
Hi @Manasa2850 
Check this issue https://github.com/eslint/eslint/issues/12473
It should be ok to disable checking function declaring for this particular file.
/*eslint no-use-before-define: ["error", { "functions": false }]*/

Reply:
Thanks for finding the eslint issue, Jacek!

I was planning to ask in the channel for help refactoring these fix this error, because I don't have the JS chops to do it myself. But if there aren't good suggestions on that today, I'd support skipping the line that raises on it.

Reply:
Thanks @Jacek-202 !!